contracts/GenArt721CoreV3.sol

```diff
@@ -631,16 +630,25 @@ contract GenArt721CoreV3 is ERC721Enumerable, IGenArt721CoreContractV3 {
             uint256 royaltyFeeByID
         )
     {
-        artistAddress = projectIdToArtistAddress[tokenIdToProjectId[_tokenId]];
-        additionalPayee = projectIdToAdditionalPayee[
-            tokenIdToProjectId[_tokenId]
-        ];
+        uint256 projectId = tokenIdToProjectId(_tokenId);
+        artistAddress = projectIdToArtistAddress[projectId];
+        additionalPayee = projectIdToAdditionalPayee[projectId];
         additionalPayeePercentage = projectIdToAdditionalPayeePercentage[
-            tokenIdToProjectId[_tokenId]
-        ];
-        royaltyFeeByID = projectIdToSecondaryMarketRoyaltyPercentage[
-            tokenIdToProjectId[_tokenId]
+            projectId
         ];
+        royaltyFeeByID = projectIdToSecondaryMarketRoyaltyPercentage[projectId];
+    }
+
+    /**
+     * @notice Derives the `projectId` for a given `_tokenId`, for convenience
+     * purposes.
+     */
+    function tokenIdToProjectId(uint256 _tokenId)
+        public
+        view
+        returns (uint256 _projectId)
+    {
+        return _tokenId - (_tokenId % ONE_MILLION);
```

Comment:
just checking bc i don't know - is this cheaper than doing `_tokenId / ONE_MILLION`?

Reply:
IIUC, they should cost the same unless the compiler is doing some goofy stuff, as the `MOD` and `DIV` opcodes both have the same gas costs associated with them: https://ethereum.org/en/developers/docs/evm/opcodes/

That said, I don't think `_tokenId / ONE_MILLION` would give us what we want actually and am now realizing that this logic as implemented doesn't either. I believe this should actually be:

```
return (_tokenId - (_tokenId % ONE_MILLION)) / ONE_MILLION; 
```

This is making me realizing we don't have proper test coverage here. Moving this PR back into draft mode and will update with fixed logic + tests.

Reply:
I was not thinking straight at all this morning `_tokenId / ONE_MILLION` is definitely what we want, and is equivalent to `(_tokenId - (_tokenId % ONE_MILLION)) / ONE_MILLION`. 

smh 🤦‍♂️ 

Thank you for flagging @grantoesterling

Reply:
why wouldn't this just be `_tokenId / 1_000_000`? (noting this because this logic is used in the in-work tokenHolderMinter)